core/primitives/src/congestion_info.rs

```diff
@@ -1,9 +1,88 @@
+/// much other shards are allowed to forward to this shard.
+const MAX_CONGESTION_OUTGOING_GAS: Gas = 2 * PGAS;
+
+/// How much memory space of all delayed and buffered receipts in a shard is
+/// considered 100% congested.
+///
+/// Memory congestion contributes to overall congestion, which reduces how much
+/// other shards are allowed to forward to this shard.
+///
+/// This threshold limits memory requirements of validators to a degree but it
+/// is not a hard guarantee.
+const MAX_CONGESTION_MEMORY_CONSUMPTION: u64 = bytesize::ByteSize::mb(1000u64).0;
+
+/// The maximum amount of gas attached to receipts a shard can forward to
+/// another shard per chunk.
+///
+/// The actual gas forwarding allowance is a linear interpolation between
+/// [`MIN_GAS_FORWARDING`] and [`MAX_GAS_FORWARDING`], or 0 if the receiver is
+/// fully congested.
+const MAX_GAS_FORWARDING: Gas = 300 * PGAS;
+
+/// The minimum gas each shard can send to a shard that is not fully congested.
+///
+/// The actual gas forwarding allowance is a linear interpolation between
+/// [`MIN_GAS_FORWARDING`] and [`MAX_GAS_FORWARDING`], or 0 if the receiver is
+/// fully congested.
+const MIN_GAS_FORWARDING: Gas = 1 * PGAS;
+
+/// How much gas the chosen allowed shard can send to a 100% congested shard.
+///
+/// This amount is the absolute minimum of new workload a congested shard has to
+/// accept every round. It ensures deadlocks are provably impossible. But in
+/// ideal conditions, the gradual reduction of new workload entering the system
+/// combined with gradually limited forwarding to congested shards should
+/// prevent shards from becoming 100% congested in the first place.
+const RED_GAS: Gas = 1 * PGAS;
+
+/// The maximum amount of gas in a chunk spent on converting new transactions to
+/// receipts.
+///
+/// The actual gas forwarding allowance is a linear interpolation between
+/// [`MIN_TX_GAS`] and [`MAX_TX_GAS`], based on the incoming congestion of the
+/// local shard. Additionally, transactions can be rejected if the receiving
+/// remote shard is congested more than [`REJECT_TX_CONGESTION_THRESHOLD`] based
+/// on their general congestion level.
+const MAX_TX_GAS: Gas = 500 * TGAS;
+
+/// The minimum amount of gas in a chunk spent on converting new transactions
+/// to receipts, as long as the receiving shard is not congested.
+///
+/// The actual gas forwarding allowance is a linear interpolation between
+/// [`MIN_TX_GAS`] and [`MAX_TX_GAS`], based on the incoming congestion of the
+/// local shard. Additionally, transactions can be rejected if the receiving
+/// remote shard is congested more than [`REJECT_TX_CONGESTION_THRESHOLD`] based
+/// on their general congestion level.
+const MIN_TX_GAS: Gas = 20 * TGAS;
+
+/// How much gas in delayed receipts a shard can tolerate before it stops all
+/// shards from accepting new transactions with the receiver set to the
+/// congested shard.
+const REJECT_TX_CONGESTION_THRESHOLD: f64 = 0.25;
```

Comment:
nit: The comment says it's measured in gas but the type and value is the congestion level.